gcc/rust/backend/rust-compile-pattern.cc

```diff
@@ -387,5 +387,51 @@ CompilePatternLet::visit (HIR::WildcardPattern &pattern)
     }
 }
 
+void
+CompilePatternLet::visit (HIR::TuplePattern &pattern)
+{
+  rust_assert (!ty->is_unit ());
```

Comment:
Note that
```rust
fn main() {
    let () = ();
}
```
is valid Rust but we cannot parse it yet. It'd be nice to have a TODO or FIXME here and open up an issue (which I can do if you want)